### ISSUE TITLE
Guard init jekyll_plus doesn't work

### ISSUE THREAD

**Issue by imathis** [imathis/guard-jekyll-plus#36]
@e2 It looks like these changes didn't work for Guard. Here's what I'm finding with the latest release of this plugin.

```
$ guard init jekyll_plus
23:33:43 - INFO - Writing new Guardfile to /Users/imathis/workspace/test/Guardfile
23:33:43 - ERROR - Could not find 'guard-jekyll_plus' gem path.
23:33:43 - ERROR - Could not load 'guard/jekyll_plus' or '~/.guard/templates/jekyll_plus' or find class Guard::Jekyll_plus
```

And of course:

```
$ guard init jekyll-plus
23:33:48 - ERROR - Could not load 'guard/jekyll-plus' or '~/.guard/templates/jekyll-plus' or find class Guard::Jekyllplus
```

I noticed that the tests do not test the command line usage, but rather interact directly with the Ruby class. I do not like mixing underscores and dashes, I'd really rather not rename the gem. Do you have a suggestion for how I can fix this?

**Comment by e2** [comment]
You can sidestep the gem name issues by using Bundler - `bundle exec init jekyll_plus` with `gem 'guard-jekyll-plus'` will load the class, and then Guard should find the class.

Bundler helps here, because if the module is loaded (i.e. no `require: false` in the `Gemfile`), Guard shouldn't care about the gem name at all.

Overall, I do want to deprecate/remove the gem name and file name handling in Guard completely. If there are any overrides necessary, they should be in pure Ruby.

Another workaround which would be "cleaner" for users is:
```bash
$ guard init Jekyll::Plus # so it will look for the gem 'guard-jekyll-plus'
```

This doesn't work yet, but if you like that solution, I'll get it working as soon as I can.

In fact, I also want to support this in an `Guardfile`:
```ruby
guard :JekyllPlus do  # or 'Jekyll::Plus' if you decide to go that route
  # (...)
end
```

As for the template - you can override the default plugin methods to get it to work, so basically you should be able to have any file working. (`guard init` is the major problem here.)

If you want to keep the convention broken: http://guides.rubygems.org/name-your-gem, there's going to be a bit more work for both you and me here - and I wouldn't go for any quickfix, because it will backfire sooner or later.

> I do not like mixing underscores and dashes

If you're willing to keep the gem name at all costs, you can stick to the convention by having the naming: `Guard::Jekyll::Plus` - and if that doesn't work, I'll get it to work.

The short story is: 
- if you follow the RubyGems convention 100% (e.g. Guard::Jekyll::Plus), I'll get Guard working ASAP
- GEM NAME - if *just* the gem name doesn't follow the convention, either you explicitly require people to use a Gemfile (so it's loaded either by Bundler or newer RubyGems) - or require the gem in the Guardfile explicitly (should work too)
- otherwise, I'd prefer to support pure Ruby solutions (module names and methods instead of file names)
- I don't want to maintain non-Gemfile workarounds for filenames that break the RubyGems convention (unless you're getting a flood of requests from users)
- I don't want to be dealing with tickets due to non-RubyGems convention in naming gone broke
- I don't want to be writing and maintaining 20-30 more tests just to support unconventional plugin naming

I've decided that I don't want to deal with gem names or file names at all (if possible) - unless it perfectly fits the Rubygems convention, or Bundler is a requirement (so that mess can be resolved in the Gemfile).

So ideally take a look at the RubyGems convention and pick something that fits the convention and whatever you want (e.g. if you want to keep the gem name, the module name would be Jekyll::Plus and a dash in file names).

The further you stray from the convention - the more work for me and you.

**Comment by e2** [comment]
If you want to keep the gem name and have it work without bundler, you'll have to expect only `guard init jekyll-plus` to work. I could support something like: `guard init JekyllPlus:guard-jekyll-plus` - but that's just madness going too far.

**Comment by e2** [comment]
Just a final note: Guard is likely broken in how it handles names anyway - which is why this is so confusing and complex. Either follow the convention or break the convention deliberately - and then I'll work out how to fix/patch Guard to make it work. Any backward-compatibility work comes afterwards (and probably worth it only if users are complaining).

**Comment by lfender6445** [comment]
where does this file get created? ruby 2 i get this:

```
~/source/lfender6445.github.io auto_pilot_integration ⚡  2.1.3
$ bundle exec guard init jekyll-plus
13:55:52 - ERROR - Could not load 'guard/jekyll-plus' or '~/.guard/templates/jekyll-plus' or find class Guard::Jekyllplus
13:55:52 - ERROR - Error is: No such file or directory @ rb_sysopen - /Users/lfender/.guard/templates/jekyll-plus
~/source/lfender6445.github.io auto_pilot_integration ⚡  2.1.3
$ ls ~/.guard/templates/
ls: /Users/lfender/.guard/templates/: No such file or directory
```

if the folder doesn't exist, the template init will fail even with bundle exec

**Comment by e2** [comment]
@imathis  - I've released Guard v2.12.5, so as soon as you release #37, you can close this issue.

NOTE: the template has changed (guard name) - previously copied Guardfile templates  won't work, so you may want to bump the major version if you like (see README changes to get the idea).

**Comment by e2** [comment]
@lfender6445 - thanks for reporting this. I've also fixed the confusing message (that area of Guard is a mess that I'll get to fixing some day...).

Copying the Guardfile is the way to go (until this is closed). Also, you'll have to update your Guardfile afterwards (to use `guard :jekyll-plus`).
